packages/api/src/routes/user.ts

```diff
@@ -341,7 +358,19 @@ router.delete(
     const provider = getFrom("query").orFail("provider", req);
     const deviceId = getFrom("query").orFail("deviceId", req);
 
-    await removeDevice(connectedUser, provider, deviceId);
+    const extraParams: DAPIParams = {
+      xTenoviApiKey: getTenoviApiKeyFrom("headers", req).optional(),
+      xTenoviClientName: getTenoviClientNameFrom("headers", req).optional(),
+    };
+
+    if (provider === PROVIDER_TENOVI) {
+      if (!extraParams.xTenoviApiKey) throw new BadRequestError("Missing x-tenovi-api-key header.");
+      if (!extraParams.xTenoviClientName) {
+        throw new BadRequestError("Missing x-tenovi-client header.");
+      }
+    }
```

Comment:
If we take the route to send `rawParams` to the provider code, could move these there as well.